python/hsfs/core/query.py

```diff
@@ -22,27 +22,44 @@
 
 class Query:
     def __init__(
-        self, feature_store_name, feature_store_id, left_feature_group, left_features
+        self,
+        feature_store_name,
+        feature_store_id,
+        left_feature_group,
+        left_features,
+        left_featuregroup_starttime=None,
+        left_featuregroup_endtime=None,
     ):
         self._feature_store_name = feature_store_name
         self._feature_store_id = feature_store_id
         self._left_feature_group = left_feature_group
         self._left_features = util.parse_features(left_features)
+        self._left_featuregroup_starttime = left_featuregroup_starttime
+        self._left_featuregroup_endtime = left_featuregroup_endtime
         self._joins = []
         self._query_constructor_api = query_constructor_api.QueryConstructorApi()
         self._storage_connector_api = storage_connector_api.StorageConnectorApi(
             feature_store_id
         )
 
-    def read(self, storage="offline", dataframe_type="default"):
+    def read(
+        self, storage="offline", dataframe_type="default", time_travel_fomat="HUDI",
+    ):
```

Comment:
Why do you need the `time_travel_format` option here? That should come from the feature group involved in the join

Reply:
That is what you do in Java

Reply:
done